server.go

```diff
@@ -57,12 +71,40 @@ func start(ctx context.Context, port int, runAcceleratedDHTClient bool, contentE
 	log.Printf("Listening on http://0.0.0.0:%d", port)
 	log.Printf("Delegated Routing API on http://127.0.0.1:%d/routing/v1", port)
 
-	http.Handle("/", server.Handler(&composableRouter{
+	mdlw := middleware.New(middleware.Config{
+		Recorder: metrics.NewRecorder(metrics.Config{Prefix: "someguy"}),
+	})
+
+	handler := server.Handler(&composableRouter{
 		providers: crRouters,
 		peers:     prRouters,
 		ipns:      ipnsRouters,
-	}))
-	return http.ListenAndServe(":"+strconv.Itoa(port), nil)
+	})
+
+	// Add CORS.
+	handler = cors.New(cors.Options{
+		AllowedOrigins: []string{"*"},
+		AllowedMethods: []string{http.MethodGet, http.MethodOptions},
+		MaxAge:         600,
+	}).Handler(handler)
+
+	// Add compression.
+	compress, err := httpcompression.DefaultAdapter()
```

Comment:
Note to myself: check if this handler adds extra buffering.
If this happen this would be bad since we would be forced to wait before using any of the response.

I think it could be possible that a writer would need to first fill it's window buffer before streaming out bytes. We can compress each entry individually and concatenate them if this ever happens (`decompress(compress(a + b)) == decompress(compress(a) + compress(b))` for gzip and zstd).